test/faktory_worker/connection_manager/server_test.exs

```diff
@@ -2,12 +2,41 @@ defmodule FaktoryWorker.ConnectionManager.ServerTest do
   use ExUnit.Case
 
   import Mox
+  import FaktoryWorker.ConnectionHelpers
 
+  alias FaktoryWorker.Random
   alias FaktoryWorker.ConnectionManager.Server
 
   setup :set_mox_global
   setup :verify_on_exit!
 
+  describe "send_command/3" do
+    test "should handle server timeouts for fetch commands" do
+      worker_connection_mox()
+
+      expect(FaktoryWorker.SocketMock, :send, fn _, _ ->
+        # sleep to make sure the fetch command takes longer
+        # than the genserver timeout specified below
+        Process.sleep(1000)
```

Comment:
Can it be less than this or does it only work at 1000?

Reply:
Yep it can be lower, though to note this sleep never lasts that long since the process that is  sleeping is shutdown as soon as the test completes. This also doesn't impact the test running time and on my machine the test runs in about 0.1 seconds.